### PR TITLE
Ability to use a custom color palette

### PR DESCRIPTION
**Pull request by HenrryEsposito**
This pull request aims to implement a way for the user to manually edit the colors of the color palette used for coloring the tags.

A new input was created on the plugin's settings screen to receive the customized colors. The user should enter the hex code of the desired colors, separating each one with commas.

image

### CODE REVIEW

**Comment by pfrankov** [comment]
Thank you for you suggestion. I'd like to keep plugin simple, so can we hide this option (maybe more options in future) in `Advanced` toggle?

**Comment by HenrryEsposito** [comment]
Hey @pfrankov !
I ended up noticing that there was an issue with my implementation. The way I set it up to ignore the palette settings and use custom colors ended up causing some bugs when the user no longer wanted to use the custom colors. I solved the problem by creating a toggle.
I hope you like how it turned out, but an advanced settings area would also be nice.

**Comment by HenrryEsposito** [comment]
Custom colors disabled:
image

Custom colors enabled:
image

File: main.ts [hunk 1]
```diff
@@ -424,76 +445,103 @@ class ColoredTagsPluginSettingTab extends PluginSettingTab {
...
+            .addToggle(toggle => toggle
+                .setValue(this.plugin.settings.enableCustomColors)
+                .onChange(async (value) => {
+                    this.plugin.settings.enableCustomColors = value;
+                    await this.plugin.saveSettings();
+					this.display();
+					this.renderPalette(paletteEl);
+                }));
+
+		if(this.plugin.settings.enableCustomColors !== true) {
```
Comment (pfrankov) [inline main.ts:466]: This is unusual condition. Maybe `if (!this.plugin.settings.enableCustomColors)` will be better?

File: main.ts [hunk 2]
```diff
@@ -424,76 +445,103 @@ class ColoredTagsPluginSettingTab extends PluginSettingTab {
...
-				dropdown.addOption(String(DEFAULT_SETTINGS.lightness), 'Default')
-					.addOptions({
-						'0': 'Dark',
-						'32': 'Medium Dark',
-						'64': 'Medium',
-						'90': 'Light',
-						'100': 'Bleach',
-					})
-					.setValue(String(this.plugin.settings.lightness))
+		if(this.plugin.settings.enableCustomColors === true) {
```
Comment (pfrankov) [inline main.ts:533]: Why do we need strict check here? Possible `if (this.plugin.settings.enableCustomColors)` will be enough

File: main.ts [hunk 3]
```diff
@@ -424,76 +445,103 @@ class ColoredTagsPluginSettingTab extends PluginSettingTab {
...
-						'128': 'Excessive',
-					})
-					.setValue(String(this.plugin.settings.chroma))
-					.onChange(async (value) => {
-						this.plugin.settings.chroma = Number(value);
-						await this.plugin.saveSettings();
-						this.renderPalette(paletteEl);
-					})
-			)
+            .setName('Enable custom colors')
```
Comment (pfrankov) [inline main.ts:455]: If it's custom it probably should go after the defaults

**Review by pfrankov — CHANGES_REQUESTED** [review]
Seems code doesn't work if no colors are provided. Could you please resolve all previous comments and double test the result?

**Comment by pfrankov** [comment]
@HenrryEsposito thank you again for the idea! I've implemented it in https://github.com/pfrankov/obsidian-colored-tags/releases/tag/5.0.0

**Pull request closed without merging**